### ISSUE TITLE
Libvirt, documentation

### ISSUE THREAD

**Issue by metal3d** [coreos/bugs#1077]
Hi
I tried for hours to understand why I cannot connect with SSH on libvirt machines deployed from your script wget https://raw.githubusercontent.com/coreos/docs/master/os/deploy_coreos_libvirt.sh

Following:
https://coreos.com/os/docs/latest/booting-with-libvirt.html

I seems there are at least 1 missing point and an error on the template.

First, SELinux machines should do this:
```
semanage fcontext -a -t virt_content_t "/var/lib/libvirt/images/coreos(/.*)"
restorecon -R /var/lib/libvirt/images/coreos
```

Second, the yaml file need a "---" between the header and the first yaml entry:
```yaml
#cloud-config
---
ssh_authorized_keys:
 - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGdByTgSVHq.......
```

Without thoses points, I was unable to correctly use CoreOS instances (ssh connexion, no discovering, and so on)

Thanks

**Comment by metal3d** [comment]
It seems the patch is not absolutly correct. Missing "(/.*)" in the semanage command, or something more. At this time I must stop the entire VM, hit the semanage commande for the entire LIBVIRT_PATH then restorecon and at last start VM.
Without this, coreos VM won't read conf-2 mount point

**Comment by crawford** [comment]
> Second, the yaml file need a "---" between the header and the first yaml entry:

What makes you think this is the case? coreos-cloudinit shouldn't care about document delimiters (in fact, Go's YAML-parsing library doesn't even support them!).

**Comment by metal3d** [comment]
I'm ok for the delimiter, it seems that I merge 2 problems in one :)

For SELinux, I just finished the correct patch - I will edit my initial post to put the right patch

```patch
diff -up ./deploy_coreos_libvirt.sh.friend ./deploy_coreos_libvirt.sh
--- ./deploy_coreos_libvirt.sh.friend	2016-01-25 14:18:15.431503583 +0100
+++ ./deploy_coreos_libvirt.sh	2016-01-25 14:17:04.572893762 +0100
@@ -51,6 +51,12 @@ for SEQ in $(seq 1 $1); do
 
         sed "s#%HOSTNAME%#$COREOS_HOSTNAME#g;s#%DISCOVERY%#$ETCD_DISCOVERY#g" $USER_DATA_TEMPLATE > $LIBVIRT_PATH/$COREOS_HOSTNAME/openstack/latest/user_data
 
+
+        if [[ selinuxenabled ]] ;then
+            echo Making SELinux conf
+            semanage fcontext -a -t virt_content_t "$LIBVIRT_PATH/$COREOS_HOSTNAME(/.*)"
+            restorecon -R "$LIBVIRT_PATH"
+        fi
         virt-install --connect qemu:///system \
                      --import \
                      --name $COREOS_HOSTNAME \
```

Without that, I cannot connect SSH on my hosts.

**Comment by kayrus** [comment]
@metal3d which selinux policy do you use? and which distro?

**Comment by kayrus** [comment]
@mjg59 can you look at this patch?

**Comment by metal3d** [comment]
@kayrus I'm using Fedora 22 with a "Enforcing" selinux

**Comment by kayrus** [comment]
@metal3d command should look like:

```sh
semanage fcontext -a -t virt_content_t "$LIBVIRT_PATH/$COREOS_HOSTNAME(/.*)?"
```

**Comment by mjg59** [comment]
I'm kind of reluctant to make assumptions about a single distribution's SELinux policy in a deployment script - there's no guarantee that other policies will use virt_content_t

**Comment by kayrus** [comment]
@mjg59 last fix tested on fedora. my debian system even doesn't boot with enabled selinux. which distro can you suggest to test selinux context?

**Comment by mjg59** [comment]
@kayrus Not quite sure what you mean - selinux policy can be entirely custom on a distribution by distribution basis.

**Comment by kayrus** [comment]
@mjg59 I mean that I was not able to test this fix in Debian. Which other popular distro with selinux can you suggest? And I'll test there.

**Comment by mjg59** [comment]
Right, I mean there's no way we can test this against all cases. We don't know what policies people might be running.

**Comment by kayrus** [comment]
@mjg59 so what would you say? should we at least merge these changes? or don't support selinux policy at all?

**Comment by mjg59** [comment]
I don't have a good answer at the moment. Merging this should certainly fix it on Fedora, but it may cause errors on other deployments. I'd recommend checking against CentOS 6, at least.

**Comment by kayrus** [comment]
@mjg59 ok. will test on centos 6 and 7

**Comment by metal3d** [comment]
@kayrus you're right for the command, it's a better option I think.
@mjg59 maybe a note on documentation can be helpful instead of merging my patch. I spent a lot of time to understand why I cannot connect to my VMs with SSH.

**Comment by kayrus** [comment]
I found a workaround. We can build iso image (https://coreos.com/os/docs/latest/config-drive.html#contents-and-format) and mount it instead of 9p fs when we detect selinux.

**Comment by metal3d** [comment]
:+1: @kayrus 
Yes, I'm using iso images on ESXi server (for a client) and I tried thoses images in libvirt, I haven't any SELinux problem that way. That's a nice idea I think.

**Comment by kayrus** [comment]
What I need to do right now is to clean-up libvirt script. And probably add possibility to run instances without sudo.

**Comment by crawford** [comment]
@kayrus any updates?

**Comment by kayrus** [comment]
@crawford try [this](https://github.com/kayrus/scripts/blob/master/deploy_coreos_cluster.sh) and let me know if it looks good for you.

It uses different workaround, generates iso image if it detects selinux, because there is no unified way to detect that on all selinux-enabled distors. Also it works fine in Ubuntu with AppArmor.

**Comment by crawford** [comment]
@kayrus seems fine.

**Comment by kayrus** [comment]
done #858
